terraform/service_redirect.tf

```diff
@@ -31,6 +31,19 @@ resource "google_service_account_iam_member" "cloudbuild-deploy-enx-redirect" {
   ]
 }
 
+resource "google_secret_manager_secret_iam_member" "server-db" {
+  for_each = toset([
+    "sslcert",
+    "sslkey",
+    "sslrootcert",
+    "password",
+  ])
+
+  secret_id = google_secret_manager_secret.db-secret[each.key].id
+  role      = "roles/secretmanager.secretAccessor"
+  member    = "serviceAccount:${google_service_account.server.email}"
+}
+
 resource "google_project_iam_member" "enx-redirect-observability" {
   for_each = toset([
     "roles/cloudtrace.agent",
```

Comment:
Github isn't letting me leave a comment, but in the `dynamic "env" {` block you need to add the configs for the cache and database.